internal/api/sms_provider/twilio.go

```diff
@@ -87,15 +87,21 @@ func (t *TwilioProvider) SendSms(phone, message, channel, otp string) (string, e
 		if isPhoneNumber.MatchString(formatPhoneNumber(sender)) {
 			sender = channel + ":" + sender
 		}
-		// Used to substitute OTP. See https://www.twilio.com/docs/content/whatsappauthentication for more details
-		contentVariables := fmt.Sprintf(`{"1": "%s"}`, otp)
+
 		// Programmable Messaging (WhatsApp) takes in different set of inputs
 		body = url.Values{
-			"To":               {receiver}, // twilio api requires "+" extension to be included
-			"Channel":          {channel},
-			"From":             {sender},
-			"ContentSid":       {t.Config.ContentSid},
-			"ContentVariables": {contentVariables},
+			"To":      {receiver}, // twilio api requires "+" extension to be included
+			"Channel": {channel},
+			"From":    {sender},
+		}
+		// For backward compatibility with old API.
+		if t.Config.ContentSid != "" {
+			// Used to substitute OTP. See https://www.twilio.com/docs/content/whatsappauthentication for more details
+			contentVariables := fmt.Sprintf(`{"1": "%s"}`, otp)
+			body.Set("ContentSid", t.Config.ContentSid)
+			body.Set("ContentVariables", contentVariables)
+		} else {
+			body.Set("Body", message)
```

Comment:
doesn't this mean that if the `ContentSid` field is specified, the channel doesn't matter and the same template will be delivered via SMS and Whatsapp?

i was under the impression that we were gonna be allowing different templates for SMS and whatsapp?

Reply:
> i was under the impression that we were gonna be allowing different templates for SMS and whatsapp?
Yes the intention is to allow different templates for SMS and WhatsApp

I think the changes here [only apply if the message is sent via a WhatsApp channel](https://github.com/supabase/gotrue/pull/1260/files#diff-9899a99332fa0703cdaf4409ba868707e6be0ec98d0931da75114b55fee331a3L85)

Reply:
oops yeah i missed out on that line